src/clr-angular/data/datagrid/datagrid-filter.ts

```diff
@@ -63,7 +63,6 @@ import { isPlatformBrowser } from '@angular/common';
           <ng-content></ng-content>
       </div>
   `,
-  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
This does seem like the root of the problem, what did changing the `set open` logic below change?

Reply:
Yes, I added this last year at the time of the first refactor for pinned columns. 
This was before the datagrid state refactor so maybe the issue I had is no longer relevant. 

Second with @gnomeontherun's question re: changed logic for the `open` getters and setters.
It seems to make more sense to me to keep it as a wrapper for `this.smartToggleService` rather than having an intermediary property that needs to be maintained.

Reply:
The main change is indeed is how `open` is tracked. The issue that I found is that the previews check:

```ts
public set open(open: boolean) {
  const boolOpen = !!open;
  if (boolOpen !== this.open) {
```
`boolOpen` and `this.open` is always the same. And the reason is that `this.open` return `return this.smartToggleService.open;` that is already changed when `set open` is called. 

For example, when the change is required we are tracking the `open` state by these lines:

```ts
    this.subs.push(
      smartToggleService.openChange.subscribe(change => {
        this.open = change;
      })
    );
```

and when it comes to executing the `if` we are passing for example `true` and `this.open` is already true cause it's set before even the subscribe method is invoked. So we never go in this `if` block, no event is emitted and so on.

Reply:
Gotcha, yes that makes sense as the problem to have the smartPopover.open value set first, then it emits the same value (so `if (boolOpen !== this.open) {` will always be false because they will equal.

By tracking the state internally we get around this problem. This is something we're going to see in other places, so we need to keep this in mind, perhaps we can come up with a better default solution later.

Reply:
We should make sure that we remove `.markForChanges()` in this component now if are removing OnPush.